tests/nn/util_test.py

```diff
@@ -437,6 +437,30 @@ def test_viterbi_decode(self):
         assert indices == [3, 2, 1]
         assert value.numpy() == 18
 
+        # Test that providing evidence results in paths containing specified tags.
+        sequence_predictions = torch.FloatTensor([[0, 0, 0, 4, 4],
+                                                  [0, 0, 0, 4, 4],
+                                                  [0, 0, 0, 4, 4],
+                                                  [0, 0, 0, 4, 4],
+                                                  [0, 0, 0, 4, 4],
+                                                  [0, 0, 0, 4, 4]])
+        # The 5th tag has a penalty for appearing sequentially
+        # or for transitioning to the 4th tag, making the best
+        # path to take the 4th tag for every label.
+        transition_matrix = torch.zeros([5, 5])
+        transition_matrix[4, 4] = -10
+        transition_matrix[4, 3] = -2
+        # The 1st, 4th and 5th tags are observed - they should be equal to
```

Comment:
I'm pretty sure you're using "4th" and "5th" here to refer to sequence indices, while above (and below!) they refer to tag indices.  Maybe always referring to tags by their values would make this easier to read?  I.e., above, "Tag 4 has a penalty..."